models/basgra/R/write.config.BASGRA.R

```diff
@@ -0,0 +1,98 @@
+#-------------------------------------------------------------------------------
+# Copyright (c) 2012 University of Illinois, NCSA.
+# All rights reserved. This program and the accompanying materials
+# are made available under the terms of the 
+# University of Illinois/NCSA Open Source License
+# which accompanies this distribution, and is available at
+# http://opensource.ncsa.illinois.edu/license.html
+#-------------------------------------------------------------------------------
+
+##-------------------------------------------------------------------------------------------------#
+##' Writes a BASGRA config file.
+##'
+##' Requires a pft xml object, a list of trait values for a single model run,
+##' and the name of the file to create
+##'
+##' @name write.config.BASGRA
+##' @title Write BASGRA configuration files
+##' @param defaults list of defaults to process
+##' @param trait.samples vector of samples for a given trait
+##' @param settings list of settings from pecan settings file
+##' @param run.id id of run
+##' @return configuration file for BASGRA for given run
+##' @export
+##' @author Istem Fer
+##-------------------------------------------------------------------------------------------------#
+write.config.BASGRA <- function(defaults, trait.values, settings, run.id) {
+
+  # find out where to write run/ouput
+  rundir <- file.path(settings$host$rundir, run.id)
+  outdir <- file.path(settings$host$outdir, run.id)
+  
+  # load default(!) BASGRA params
+  load(system.file("BASGRA_params.Rdata",package = "PEcAn.BASGRA"))
+  run_params <- default_params
```

Comment:
Personally, I really don't like `.Rdata` files. But, you might consider using `PEcAn.utils::load_local` to instead read this into a list and avoid accidentally clobbering other objects in the current namespace (like `load` does by default).

```r
run_params <- PEcAn.utils::load_local(system.file(...))$default_params
```